scripts/ci-entrypoint.sh

```diff
@@ -132,91 +132,104 @@ create_cluster() {
-    do
-        # Wait for the kubeadm-config configmap to exist.
-        sleep 2
-    done
-    "${KUBECTL}" get configmap kubeadm-config --namespace=kube-system -o yaml \
-    | sed 's/namespace: kube-system/namespace: calico-system/' \
-    | "${KUBECTL}" create -f -
-
-    # install Calico CNI
-    echo "Installing Calico CNI via helm"
-    if [[ "${CIDR0}" =~ .*:.* ]]; then
-        echo "Cluster CIDR is IPv6"
-        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-ipv6/values.yaml"
-        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
-    elif [[ "${CIDR1}" =~ .*:.* ]]; then
-        echo "Cluster CIDR is dual-stack"
-        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-dual-stack/values.yaml"
-        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0},installation.calicoNetwork.ipPools[1].cidr=${CIDR1}"
-    else
-        echo "Cluster CIDR is IPv4"
-        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico/values.yaml"
-        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
-    fi
-
-    "${HELM}" repo add projectcalico https://projectcalico.docs.tigera.io/charts
-    "${HELM}" install calico projectcalico/tigera-operator -f "${CALICO_VALUES_FILE}" --set-string "${CIDR_STRING_VALUES}" --namespace tigera-operator --create-namespace
-
-    # install cloud-provider-azure components, if using out-of-tree
-    if [[ -n "${TEST_CCM:-}" ]]; then
-        CLOUD_CONFIG="/etc/kubernetes/azure.json"
-        CONFIG_SECRET_NAME=""
-        ENABLE_DYNAMIC_RELOADING=false
-        if [[ -n "${LOAD_CLOUD_CONFIG_FROM_SECRET:-}" ]]; then
-            CLOUD_CONFIG=""
-            CONFIG_SECRET_NAME="azure-cloud-provider"
-            ENABLE_DYNAMIC_RELOADING=true
-            copy_secret
+        # install Calico CNI
+        echo "Installing Calico CNI via helm"
+        if [[ "${CIDR0}" =~ .*:.* ]]; then
+            echo "Cluster CIDR is IPv6"
+            CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-ipv6/values.yaml"
+            CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
+        elif [[ "${CIDR1}" =~ .*:.* ]]; then
+            echo "Cluster CIDR is dual-stack"
+            CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-dual-stack/values.yaml"
+            CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0},installation.calicoNetwork.ipPools[1].cidr=${CIDR1}"
+        else
+            echo "Cluster CIDR is IPv4"
+            CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico/values.yaml"
+            CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
         fi
 
-        CCM_CLUSTER_CIDR="${CIDR0}"
-        if [[ -n "${CIDR1}" ]]; then
-            CCM_CLUSTER_CIDR="${CIDR0}\,${CIDR1}"
+        "${HELM}" repo add projectcalico https://projectcalico.docs.tigera.io/charts || continue
```

Comment:
Not this PR's fault, but these `helm` commands lead to these warnings:

```
Installing Calico CNI via helm
Cluster CIDR is IPv4
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
"projectcalico" has been added to your repositories
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
NAME: calico
```

Presumably we can fix that with a `chmod`. @jackfrancis do you want to add that here or should I open an issue/PR?

Reply:
Actually can you add a note to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3126

I think we should fix that there